src/pages/index.js

```diff
@@ -0,0 +1,309 @@
+  setButtonText(submitBtn, true, "Saving...", "Save");
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = editModalNameInput.value;
+      profileDescription.textContent = editModalDescriptionInput.value;
+      closeModal(editProfileModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false, "Saving...", "Save");
+    });
+}
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true, "Saving...", "Save");
+  const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
+  api
+    .addCard(inputValues)
+    .then((data) => {
+      const cardElement = getCardElement(data);
+      cardsList.prepend(cardElement);
+      disableButton(cardSubmitButton, settings);
+      evt.target.reset();
+      closeModal(cardModal);
+    })
+    .catch(console.error);
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true, "Saving...", "Save");
+  avatarSubmitButton.disabled = true;
+  api
+    .editUserAvatar(avatarLinkInput.value)
+    .then((data) => {
+      profileAvatar.src = data.avatar;
+      closeModal(avatarModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setTimeout(() => {
+      avatarSubmitButton.disabled = false;
+      }, 1000);
+    });
+}
+
+avatarLinkInput.addEventListener("input", () => {
+    disableButton(avatarSubmitButton, settings);
+});
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true, "Deleting...", "Delete");
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(deleteModal);
+    })
+    .catch(console.error);
+}
+
+function handleDeleteCard(cardElement, cardID) {
+  selectedCard = cardElement;
+  selectedCardId = cardID;
+  openModal(deleteModal);
+}
+
+deleteModalCloseButton.addEventListener("click", () => {
+  closeModal(deleteModal);
+});
+
+deleteModalCancelButton.addEventListener("click", () => { 
+  closeModal(deleteModal);
+});
+
+profileEditButton.addEventListener("click", () => {
+  editModalNameInput.value = profileName.textContent;
+  editModalDescriptionInput.value = profileDescription.textContent;
+  resetValidation(
+    editFormElement,
+    [editModalNameInput, editModalDescriptionInput],
+    {
+      inputErrorClass: "modal__input_type_error",
+      errorClass: "modal__error",
+    }
+  );
+  openModal(editProfileModal);
+});
+editProfileCloseButton.addEventListener("click", () => {
+  closeModal(editProfileModal);
+});
+cardModalButton.addEventListener("click", () => {
+  cardNameInput.value = "";
+  cardLinkInput.value = "";
+  resetValidation(cardForm, [cardNameInput, cardLinkInput], { 
```

Comment:
`resetValidation` is for clearing possible validation errors when you open the **profile** form, because you insert **valid** data into its inputs.  


You don't need for the card form